File: backend/src/main/java/me/havooc/poligon/model/Announcement.java
/**
 * @Author Łukasz Piętka (FUT 2025)
 */
package me.havooc.poligon.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Announcement {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(length = 4096)
    private String text;
    @ManyToOne
    private Crew author;
    @CreationTimestamp
    private LocalDateTime publishDate;
    @UpdateTimestamp
    private LocalDateTime lastModified;
}

File: frontend/lib/model/announcement.dart
import 'crew.dart';

/// Author: Łukasz Piętka (FUT 2025)
class Announcement {
  final int id;
  final String text;
  final Crew author;
  final DateTime publishDate;
  final DateTime? lastModified;

  Announcement({
    this.id = -1,
    required this.text,
    required this.author,
    required this.publishDate,
    this.lastModified,
  });
}